modules/fs2/src/main/scala/higherkindness/mu/rpc/internal/client/fs2/calls.scala

```diff
@@ -16,65 +16,77 @@
 
 package higherkindness.mu.rpc.internal.client.fs2
 
-import fs2.Stream
-
 import cats.data.Kleisli
-import cats.effect.ConcurrentEffect
-import cats.syntax.applicative._
-import cats.syntax.flatMap._
-import cats.syntax.functor._
+import cats.effect.Async
+import cats.effect.std.Dispatcher
+import cats.syntax.all._
+import fs2.Stream
+import fs2.grpc.client.{ClientOptions, Fs2ClientCall}
 import higherkindness.mu.rpc.internal.client.tracingKernelToHeaders
 import io.grpc.{CallOptions, Channel, Metadata, MethodDescriptor}
-import org.lyranthe.fs2_grpc.java_runtime.client.Fs2ClientCall
 import natchez.Span
 
 object calls {
 
-  def unary[F[_]: ConcurrentEffect, Req, Res](
+  def unary[F[_]: Async, Req, Res](
       request: Req,
       descriptor: MethodDescriptor[Req, Res],
       channel: Channel,
       options: CallOptions,
       headers: Metadata = new Metadata()
-  ): F[Res] =
-    Fs2ClientCall[F](channel, descriptor, options)
-      .flatMap(_.unaryToUnaryCall(request, headers))
+  ): F[Res] = {
+    Dispatcher[F].use { disp =>
+      Fs2ClientCall[F]
+        .apply(channel, descriptor, disp, clientOptions(options))
+        .flatMap(_.unaryToUnaryCall(request, headers))
+    }
+  }
 
-  def clientStreaming[F[_]: ConcurrentEffect, Req, Res](
+  def clientStreaming[F[_]: Async, Req, Res](
       input: Stream[F, Req],
       descriptor: MethodDescriptor[Req, Res],
       channel: Channel,
       options: CallOptions,
       headers: Metadata = new Metadata()
   ): F[Res] =
-    Fs2ClientCall[F](channel, descriptor, options)
-      .flatMap(_.streamingToUnaryCall(input, headers))
+    Dispatcher[F].use { disp =>
+      Fs2ClientCall[F](channel, descriptor, disp, clientOptions(options))
+        .flatMap(_.streamingToUnaryCall(input, headers))
+    }
 
-  def serverStreaming[F[_]: ConcurrentEffect, Req, Res](
+  def serverStreaming[F[_]: Async, Req, Res](
       request: Req,
       descriptor: MethodDescriptor[Req, Res],
       channel: Channel,
       options: CallOptions,
       headers: Metadata = new Metadata()
   ): F[Stream[F, Res]] =
     Stream
-      .eval(Fs2ClientCall[F](channel, descriptor, options))
-      .flatMap(_.unaryToStreamingCall(request, headers))
-      .pure[F]
+      .resource(Dispatcher[F])
+      .flatMap { disp =>
+        Stream
+          .eval(Fs2ClientCall[F](channel, descriptor, disp, clientOptions(options)))
+          .flatMap(_.unaryToStreamingCall(request, headers))
+      }
+      .pure[F] // Why is this F[Stream[F, A]]? That type makes no sense.
```

Comment:
I believe it's because of the integration with https://github.com/tpolecat/natchez, @cb372 might confirm it